src/client/Microsoft.Identity.Client/Internal/Broker/BrokerSilentRequest.cs

```diff
@@ -58,6 +58,12 @@ private async Task<MsalTokenResponse> SendAndVerifyResponseAsync()
 
         internal void CreateRequestParametersForBroker()
         {
+            if (string.IsNullOrEmpty(_silentParameters.LoginHint) && string.IsNullOrEmpty(_silentParameters.Account?.HomeAccountId?.Identifier))
+            {
+                throw new MsalUiRequiredException(MsalError.NoUsernameOrAccountIDProvidedForSilentAndroidBrokerAuthentication, MsalErrorMessage.NoUPNOrAccountIDForSilentBrokerAuth);
+            }
```

Comment:
Is this ensuring that there is no null ref exception thrown in iOS as well as Android? and thus people will now get this instead of a null ref? Could this be thrown for other reasons? If so has iOS been fully regression tested?

Reply:
This is only for android. iOS does not have a silent broker flow and it simply checks the cache.

Reply:
This exception can also be thrown if the account is not present in the cache